R/render.r

```diff
@@ -301,3 +303,64 @@ check_made_by <- function(first) {
   if (length(first) == 0L) return(FALSE)
   grepl("<!-- Generated by pkgdown", first, fixed = TRUE)
 }
+
+pkgdown_footer <- function(data, pkg) {
+  footer_components <- list(
+    authors = footer_authors(data),
+    pkgdown = footer_pkgdown(data)
+  )
+
+  # footer left
+  left_structure <- pkg$meta$footer$left$structure %||% c("authors")
+
+  left_components <- utils::modifyList(
+    footer_components,
+    purrr::map(pkg$meta$footer$left$components, markdown_text2) %>%
+      set_names(names(pkg$meta$footer$left$components))
+  )
+
+  missing <- setdiff(left_structure, names(left_components))
+
+  if (length(missing) > 0) {
+    abort_missing(missing, where = c("footer", "left", "components"), pkg)
+  }
+
+  left_final_components <- purrr::compact(
```

Comment:
Is the compact in the wrong place? `paste0()` will return a string which compact won't affect.